packages/experimental/Shimmer/src/Shimmer.tsx

```diff
@@ -0,0 +1,97 @@
+  displayName: shimmerName,
+  ...stylingSettings,
+  slots: {
+    root: Svg,
+    rect: Rect,
+    circle: Circle,
+    image: Image,
+  },
+  render: (props: ShimmerProps, useSlots: UseSlots<ShimmerType>) => {
+    const Slots = useSlots(props);
+    let AnimatedLinearGradient = Animated.createAnimatedComponent(LinearGradient);
+    const tokens = useStyling(props).root;
+
+    const memoizedShimmerData = useMemo(
+      () => ({
+        gradientTintColor: props.gradientTintColor ? props.gradientTintColor : tokens['gradientTintColor'],
+        shimmerTintColor: props.shimmerTintColor ? props.shimmerTintColor : tokens['shimmerTintColor'],
+        containerWidth: props.width ? props.width : tokens['width'],
+        containerHeight: props.height ? props.height : tokens['height'],
+        toValue: props.toValue ? props.toValue : tokens['toValue'],
+        delay: props.delay ? props.delay : tokens['delay'],
+        duration: props.duration ? props.duration : tokens['duration'],
+        angle: props.angle ? props.angle : tokens['angle'],
+        gradientOpacity: tokens['gradientOpacity'],
+      }),
+      [props.gradientTintColor, props.shimmerTintColor, props.width, props.height, props.toValue, props.delay, props.duration, props.angle, props.gradientOpacity],
+    );
+
+    let startValue = useShimmerAnimation(memoizedShimmerData);
+
+    return (rest: ShimmerProps) => {
+      const { circle, rect, uri, elements, ...mergedProps } = mergeProps(props, rest);
+      if (elements) {
+        var rows = [];
+        for (var i = 0; i < elements.length; i++) {
+          const element = elements[i];
+          if (element.type == ShimmerElementType.rect) {
+            rows.push(<Slots.rect key={i} width={element.width} height={element.height} x={element.xPos} y={element.yPos} rx={element.borderRadius} ry={element.borderRadius} />);
+          } else if (element.type == ShimmerElementType.circle) {
+            rows.push(<Slots.circle key={i} r={element.height / 2} cx={element.xPos} cy={element.yPos} />);
+          }
+        }
+      }
+      return (
+        <Slots.root {...mergedProps}>
+          <Defs>
+            <AnimatedLinearGradient id="gradient" x1={startValue} y1={memoizedShimmerData.angle} x2="-1" y2="-1" >
+              <Stop offset="10%" stopColor={uri ? memoizedShimmerData.gradientTintColor : memoizedShimmerData.shimmerTintColor} stopOpacity={uri ? "0" : "1"} />
+              <Stop offset="20%" stopColor={memoizedShimmerData.gradientTintColor} stopOpacity={memoizedShimmerData.gradientOpacity} />
+              <Stop offset="30%" stopColor={uri ? memoizedShimmerData.gradientTintColor : memoizedShimmerData.shimmerTintColor} stopOpacity={uri ? "0" : "1"} />
+            </AnimatedLinearGradient>
+          </Defs>
+          {uri && <Slots.image href={props.uri} />}
+          <ClipPath id="shimmerView">
+            {rows}
+            {rect && <Slots.rect width={props.rect.width} x={props.rect.x} y={props.rect.y} />}
```

Comment:
Yeah, I thought it'd be cleaner for the client to construct a group view with an array, also I believe that's how Fluent web does it too.

Reply:
Ah, I see that [FluentUI React Shimmer](https://developer.microsoft.com/en-us/fluentui#/controls/web/shimmer) does it with the props, and [FluentUI React Northstar Skeleton](https://fluentsite.z22.web.core.windows.net/0.54.0/components/skeleton/definition) does it with children. That's probably why I got slightly confused. Thanks!

Reply:
Thanks for sharing the northstar version, I didn't know about it! One thing I like about using props is that the sub-elements are reusable on the client side, kinda like the example I have in the test page, (not sure if that's possible with children). I have a feeling that the northstar version should be the gold standard we should follow though...

Reply:
Reusable in that you can use the same element in multiple shimmers? Or that You can render an individual element? I think there's ways to do both. I also agree that Northstar is probably the thing to base off of, but I get that's a fairly big API change too and probably out of scope if we already like our props design

Reply:
The former. I'm not sure what the advantages of using children are aside from readability, I'm just leaning towards props for now until we have stronger reasons to do otherwise.

Reply:
I think the main benefit is clients have stronger control over the children, where they can style each child individually / layout circles and lines and rects how they see fit. 
It sounds like we weren't going for that with this Shimmer, though I still think it's a useful followup in the future, if it comes up as a client need.